docs/source/graph-construction/gs-processing/gpartition/sagemaker.rst

```diff
@@ -0,0 +1,133 @@
+2. **DEVICE_TYPE**, is the intended device type of the to-be built Docker image. Please specify ``cpu`` for building CPU-compatible images for partition job.
+3. **IMAGE_NAME** (optional), is the assigned name of the to-be built Docker image. Default is ``graphstorm``.
+
+.. warning::
+    In order to upload the GraphStorm SageMaker Docker image to Amazon ECR, users need to define the <IMAGE_NAME> to include the ECR URI string, **<AWS_ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com/**, e.g., ``888888888888.dkr.ecr.us-east-1.amazonaws.com/graphstorm``.
+
+4. **IMAGE_TAG** (optional), is the assigned tag name of the to-be built Docker image. Default is ``sm-<DEVICE_TYPE>``,
+   that is, ``sm-gpu`` for GPU images, ``sm-cpu`` for CPU images.
+
+Once the ``build_docker_sagemaker.sh`` command completes successfully, there will be a Docker image, named ``<IMAGE_NAME>:<IMAGE_TAG>``,
+such as ``888888888888.dkr.ecr.us-east-1.amazonaws.com/graphstorm:sm-gpu``, in the local repository, which could be listed by running:
+
+.. code-block:: bash
+
+    docker image ls
+
+.. _upload_sagemaker_docker:
+
+Step 2: Upload Docker Images to Amazon ECR Repository
+.......................................................
+Because SageMaker relies on Amazon ECR to access customers' own Docker images, users need to upload Docker images built in the Step 1 to their own ECR repository.
+
+The following command will authenticate the user account to access to user's ECR repository via AWS CLI.
+
+.. code-block:: bash
+
+    aws ecr get-login-password --region <REGION> | docker login --username AWS --password-stdin <AWS_ACCOUNT_ID>.dkr.ecr.<REGION>.amazonaws.com
+
+Please replace the `<REGION>` and `<AWS_ACCOUNT_ID>` with your own account information and be consistent with the values used in the **Step 1**.
+
+In addition, users need to create an ECR repository at the specified `<REGION>` with the name as `<IMAGE_NAME>` **WITHOUT** the ECR URI string, e.g., ``graphstorm``.
+
+And then use the below command to push the built GraphStorm Docker image to users' own ECR repository.
+
+.. code-block:: bash
+
+    docker push <IMAGE_NAME>:<IMAGE_TAG>
+
+Please replace the `<IMAGE_NAME>` and `<IMAGE_TAG>` with the actual Docker image name and tag, e.g., ``888888888888.dkr.ecr.us-east-1.amazonaws.com/graphstorm:sm-gpu``.
+
+Launch the partition job on Amazon SageMaker
+---------------------------------------------
+
+For this example we'll use a same size SageMaker cluster with 2 ``ml.t3.xlarge`` instances
+as in :ref:`GSProcessing Example<gsp-examining-output>`. In this tutorial, we assume the data is already ready on AWS S3 bucket.
+For large graph, users can select larger instance type and more instances in number.
+
+Install dependencies
+.....................
+To run GraphStorm with the Amazon SageMaker service, users should set up an instance with the SageMaker library installed and GraphStorm's SageMaker tools copied.
```

Comment:
No need to set up an entire instance just to run this, just need a Python env with SageMaker, and configured AWS credentials

Reply:
I change the description here.